src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorsShouldHaveSymmetricalOverloads.cs

```diff
@@ -26,13 +31,103 @@ public abstract class OperatorsShouldHaveSymmetricalOverloadsAnalyzer : Diagnost
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182356.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterSymbolAction(symbolAnalysisContext =>
+            {
+                var namedType = (INamedTypeSymbol)symbolAnalysisContext.Symbol;
+
+                // NOTE(cyrusn): We use the C# syntax when reporting diagnostics for these issues.
+                // That's what the old FxCop rule did so it doesn't seem like a bug deal.
+                CheckOperators(symbolAnalysisContext, namedType, WellKnownMemberNames.EqualityOperatorName, WellKnownMemberNames.InequalityOperatorName, "==", "!=");
+                CheckOperators(symbolAnalysisContext, namedType, WellKnownMemberNames.GreaterThanOperatorName, WellKnownMemberNames.LessThanOperatorName, ">", "<");
+                CheckOperators(symbolAnalysisContext, namedType, WellKnownMemberNames.GreaterThanOrEqualOperatorName, WellKnownMemberNames.LessThanOrEqualOperatorName, ">=", "<=");
+            }, SymbolKind.NamedType);
+        }
+
+        private void CheckOperators(
+            SymbolAnalysisContext analysisContext, INamedTypeSymbol namedType,
+            string memberName1, string memberName2,
+            string opName1, string opName2)
+        {
+            var operators1 = namedType.GetMembers(memberName1);
```

Comment:
I'm not certain i understand this.  I want to deal with the arrays and i want to work if there are multiple overloads of the operators.  you might have "operator == (A, B)" and "operator == (A, C)" etc. etc.